### PR TITLE
Feat/mobile 2897 update challenges api

### PR DESCRIPTION
**Pull request by arturdm**

### CODE REVIEW

File: sdk/src/main/java/com/meniga/sdk/models/challenges/MenigaChallenge.java [hunk 1]
```diff
@@ -425,12 +427,13 @@ public void merge(MenigaChallenge other) {
 	}
 
 	/**
-	 * Gets all challenges, expired, suggested and accepted alike but not expired ones
+	 * Gets all challenges - expired, suggested, accepted and disabled.
 	 *
 	 * @return A task that includes all the challenge objects that match the default parameters
 	 */
 	public static Result<List<MenigaChallenge>> fetch() {
```
Comment (agustk) [inline sdk/src/main/java/com/meniga/sdk/models/challenges/MenigaChallenge.java]: Should we keep the original functionality in a deprecated method?

File: sdk/src/main/java/com/meniga/sdk/models/challenges/MenigaChallenge.java [hunk 2]
```diff
@@ -468,9 +471,23 @@ public void onFinished(MenigaChallenge result, boolean failed) {
 	 * Gets all challenges, both new ones and accepted ones, also excluded ones if includeExpired is set to true
```
Comment (agustk) [inline sdk/src/main/java/com/meniga/sdk/models/challenges/MenigaChallenge.java]: Maybe add a comment explaining the effects of the new parameter, which is set to true.

File: sdk/src/main/java/com/meniga/sdk/models/challenges/MenigaChallenge.java [hunk 3]
```diff
@@ -525,6 +542,14 @@ public void onFinished(MenigaChallenge result, boolean failed) {
 		});
 	}
 
+	public Result<Void> disable() {
```
Comment (agustk) [inline sdk/src/main/java/com/meniga/sdk/models/challenges/MenigaChallenge.java]: Documentation missing?

File: sdk/src/main/java/com/meniga/sdk/webservices/PersistenceDelegate.java [hunk 4]
```diff
@@ -681,37 +690,45 @@ public void onFailure(Call<E> call, Throwable t) {
...
-		return persist(req, getClient(Service.CHALLENGES).updateChallenge(req.id.toString(), req));
+		return persist(req, getService(ChallengeService.class).updateChallenge(req.id.toString(), req));
 	}
 
 	public Result<List<MenigaChallenge>> getChallengeHistory(GetChallengeHistory req) {
 		if (provider.hasKey(req)) {
 			return createTask(provider.fetch(req));
 		}
-		return persist(req, getClient(Service.CHALLENGES).getChallengeHistory(req.id.toString(), req.toQueryMap()));
+		return persist(req, getService(ChallengeService.class).getChallengeHistory(req.id.toString(), req.toQueryMap()));
```
Comment (agustk) [inline sdk/src/main/java/com/meniga/sdk/webservices/PersistenceDelegate.java]: Does this handle different endpoint URIs for different services?

Comment (arturdm) [reply sdk/src/main/java/com/meniga/sdk/webservices/PersistenceDelegate.java]: Yup, it should work exactly the same as with `MenigaAPI`. The logic behind this is located in `WebClient.kt`.

File: sdk/src/main/java/com/meniga/sdk/models/challenges/MenigaChallenge.java (hunk 1 above)
Comment (arturdm) [reply sdk/src/main/java/com/meniga/sdk/models/challenges/MenigaChallenge.java]: This method description was misleading - it also included expired ones (passing includeExpired: true).

But a new parameter was introduced - `includeDisabled` so this method returns all the challanges except disabled ones.

I guess this one should retrieve all the challanges by default?

File: sdk/src/main/java/com/meniga/sdk/models/challenges/MenigaChallenge.java (hunk 2 above)
Comment (arturdm) [reply sdk/src/main/java/com/meniga/sdk/models/challenges/MenigaChallenge.java]: You're right that this one should work like before as it's deprecated now.

File: sdk/src/main/java/com/meniga/sdk/models/challenges/MenigaChallenge.java (hunk 1 above)
Comment (agustk) [reply sdk/src/main/java/com/meniga/sdk/models/challenges/MenigaChallenge.java]: Yes I think so, the fetch with no parameters should get all items.

**Pull request merged**